liblocation/src/main/java/com/mapbox/android/core/location/LocationPendingIntentProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.Random;
+
+class LocationPendingIntentProvider {
+
+  private final LocationPendingIntent locationPendingIntent;
+
+  LocationPendingIntentProvider(Context context, GoogleLocationEngine googleLocationEngine) {
+    locationPendingIntent = buildIntent(context, googleLocationEngine);
+  }
+
+  LocationPendingIntent intent() {
+    return locationPendingIntent;
+  }
+
+  static LocationPendingIntent buildIntent(Context context, GoogleLocationEngine googleLocationEngine) {
+    Intent intent = new Intent("GoogleLocationEngineBroadcast-" + googleLocationEngine.toString());
+
+    PendingIntent broadcastPendingIntent = PendingIntent.getBroadcast(context, generateRequestCode(), intent, 0);
+    return new LocationBroadcastPendingIntent(broadcastPendingIntent);
+  }
+
+  private static int generateRequestCode() {
+    Random random = new Random();
+    return random.nextInt();
```

Comment:
We need to statically keep track of all of the registered request codes, otherwise, there's a chance that we will overwrite one of the requests when creating a new instance of the engine.